ttnn/cpp/ttnn/operations/core/compute_kernel/compute_kernel_config.cpp

```diff
@@ -117,4 +119,27 @@ std::tuple<MathFidelity, bool, bool, bool, bool> get_compute_kernel_config_args(
     return std::make_tuple(math_fidelity, math_approx_mode, fp32_dest_acc_en, packer_l1_acc, dst_full_sync_en);
 }
 
+uint32_t get_dest_reg_count(const DeviceComputeKernelConfig& compute_kernel_config) {
+    uint32_t available_reg_count = MAX_DEST_REG_COUNT;
+    std::visit(
+        [&](auto&& compute_kernel_config) {
+            using T = std::decay_t<decltype(compute_kernel_config)>;
+            if constexpr (std::is_same_v<T, GrayskullComputeKernelConfig>) {
+                available_reg_count /= 2; // fp32_dest_acc and dst_full_sync disabled for GS
```

Comment:
Where is dst_full_sync for GS disabled ? I made the changes for all the arcs though

Reply:
Line 105 above:
```
dst_full_sync_en = false;
```

Reply:
You do set it on line 32 though. I try to make the check for it common.

Reply:
I mean do you need full dst sync for grayskull ? AFAIK all the code is there to support it

Reply:
It would be useful to have before we work on https://github.com/tenstorrent/tt-metal/issues/13485